dsc_lib/src/dscresources/command_resource.rs

```diff
@@ -815,26 +816,94 @@ pub fn log_stderr_line<'a>(process_id: &u32, trace_line: &'a str) -> &'a str
+                include_target = false;
+                ""
+            };
+            let line_number = if let Some(l) = trace_object.line_number { l } else {
+                include_target = false;
+                0
+            };
+            let trace_message = if include_target {
+                format!("Process {process_id}: {target}: {line_number}: {}", trace_object.fields.message)
+            } else {
+                format!("Process {process_id}: {}", trace_object.fields.message)
+            };
+            match trace_object.level {
+                TraceLevel::Error => {
+                    error!(trace_message);
+                },
+                TraceLevel::Warning => {
+                    warn!(trace_message);
+                },
+                TraceLevel::Info => {
+                    info!(trace_message);
+                },
+                TraceLevel::Debug => {
+                    debug!(trace_message);
+                },
+                TraceLevel::Trace => {
+                    trace!(trace_message);
+                },
+            }
+        }
+        else if let Ok(json_obj) = serde_json::from_str::<Value>(trace_line) {
             if let Some(msg) = json_obj.get("Error") {
-                error!("Process id {process_id} : {}", msg.as_str().unwrap_or_default());
+                error!("Process {process_id}: {}", msg.as_str().unwrap_or_default());
             } else if let Some(msg) = json_obj.get("Warning") {
-                warn!("Process id {process_id} : {}", msg.as_str().unwrap_or_default());
+                warn!("Process {process_id}: {}", msg.as_str().unwrap_or_default());
             } else if let Some(msg) = json_obj.get("Info") {
-                info!("Process id {process_id} : {}", msg.as_str().unwrap_or_default());
+                info!("Process {process_id}: {}", msg.as_str().unwrap_or_default());
             } else if let Some(msg) = json_obj.get("Debug") {
-                debug!("Process id {process_id} : {}", msg.as_str().unwrap_or_default());
+                debug!("Process {process_id}: {}", msg.as_str().unwrap_or_default());
             } else if let Some(msg) = json_obj.get("Trace") {
-                trace!("Process id {process_id} : {}", msg.as_str().unwrap_or_default());
+                trace!("Process {process_id}: {}", msg.as_str().unwrap_or_default());
             } else {
                 // the line is a valid json, but not one of standard trace lines - return it as filtered stderr_line
+                trace!("Process {process_id}: {trace_line}");
                 return trace_line;
             };
         } else {
             // the line is not a valid json - return it as filtered stderr_line
+            trace!("Process {process_id}: {}", trace_line);
             return trace_line;
         }
     };
 
     ""
 }
+
+#[derive(PartialEq, Eq, Deserialize)]
+enum TraceLevel {
```

Comment:
We already have `pub enum TraceLevel` in `dsc\src\args.rs`
Can we have just one of these?

Reply:
Since `dsc` already reaches deep into `dsc_lib` for some things, we can have `dsc` use the one in `dsc_lib`, the only thing that's not ideal is that `dsc_lib` now will have to depend on `clap` crate